lib/processor/download.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const url = require('url');
+const path = require('path');
+const fs = require('fs');
+const Q = require('q');
+const http = require('follow-redirects').http;
+const https = require('follow-redirects').https;
+const s3 = new (require('aws-sdk')).S3();
+
+// get a remote file source
+module.exports = (uploadUrl) => {
+  let fileName = path.basename(url.parse(uploadUrl || 'unknown').pathname);
+  let writeStream = fs.createWriteStream(`/tmp/${fileName}`)
+  let result = {name: fileName, localPath: `/tmp/${fileName}`};
+
+  return Q.Promise((resolve, reject) => {
+    const rejectNoRetry = err => {
+      err.noRetry = true;
+      reject(err);
+    }
+
+    writeStream.on('error', err => reject(err));
+    writeStream.on('close', () => {
+      if (result.contentLength && +result.contentLength !== writeStream.bytesWritten) {
+        reject(new Error(`Expected ${result.contentLength} bytes from ${uploadUrl} - got ${writeStream.bytesWritten}`));
+      } else {
+        resolve(result);
+      }
+    });
+
+    if (!uploadUrl) {
+      rejectNoRetry(new Error('No url set for event'));
+    } else if (uploadUrl.startsWith('s3://')) {
+      let m = uploadUrl.match(/s3:\/\/([^\/]+)\/(.*)/);
+      result.s3Bucket = m[1];
+      result.s3Key = m[2];
+      s3.getObject({Bucket: m[1], Key: m[2]})
+        .createReadStream()
+        .on('error', err => rejectNoRetry(err))
```

Comment:
in your comment on this PR it sounded like we do want to retry S3 upload errors? 
```
Only reject errors we want to retry
- HTTP downloads with status >= 500
- S3 uploads
```

Reply:
Yeah, but this is the download!

Also, I'm actually not sure how to differentiate "not found / not authorized / timeout" errors returned by S3 here.  I'm hoping that by throwing/logging them, I can figure that out and retry only the timeouts here at some point.

Reply:
oops!